CMakeLists.txt

```
@@ -0,0 +1,72 @@
+###############################################################################
+#
+#  Copyright (c) 2022 Contributors to the Eclipse Foundation
+#
+#  See the LICENSE file(s) distributed with this work for additional
+#  information regarding copyright ownership.
+#
+#  This program and the accompanying materials are made available under the
+#  terms of the Eclipse Public License 2.0 
+#  which is available at http://www.eclipse.org/legal/epl-2.0
+#  and the Eclipse Distribution License v. 1.0
+#  available at http://www.eclipse.org/org/documents/edl-v10.php
+#
+#  SPDX-License-Identifier: EPL-2.0
+#
+#  Contributors:
+#     Jimmy Björklund  - initial version
+#     Achim Kraus      - minor fixes
+#
+###############################################################################
+
+cmake_minimum_required(VERSION 3.5)
+
+project(tinydtls)
+
+include (AutoConf.cmake)
+
+option(BUILD_SHARED_LIBS "Link using shared libs" OFF)
+option(make_tests "Make test programs" OFF)
+
+if(NOT PLATFORM)
+   set(PLATFORM "posix" CACHE STRING "Choose platform." FORCE)
+   set_property(CACHE PLATFORM PROPERTY STRINGS "contiki" "espidf" "posix" "riot")
+endif()
+
+set(PACKAGE_NAME "tinydtls")
+set(PACKAGE_VERSION "0.8.6" )
+
+option(DTLS_ECC "disable/enable support for TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8" ON )
+option(DTLS_PSK "disable/enable support for TLS_PSK_WITH_AES_128_CCM_8" ON)
+
+configure_file(dtls_config.h.cmake.in dtls_config.h )
+
+add_library(tinydtls)

```

For my use cases, I need tinydtls compiled to a shared/dynamically linked library like so:

```suggestion
add_library(tinydtls SHARED)
```

Could there be an option added that allows for enabling this mode of compilation?

---

AFAIK, using

```
cmake -DBUILD_SHARED_LIBS=ON  ../tinydtls
```
will do that without the request change. I added this to the CMake section in the README.

If possible, please test, if that "works for you". 

(By the way, I need to test the install stuff, I skipped that in order to get faster feddbacks.)

---

I just tested it and generating the shared library works fine :) However, for some reason there are three files which are being generated:

```
libtinydtls.so (symbolic link to libtinydtls.so.0)
libtinydtls.so.0 (symbolic link to libtinydtls.so.0.8.6)
libtinydtls.so.0.8.6
```

Using `libtinydtls.so.0.8.6` (after renaming it) works as intended, however, it would probably be better if only `libtinydtls.so` was generated as an actual file?

---

> Ah, that's the way, unix does versioning for shared libraries with symbolic-links. Should I add a option to remove that?

Oh, sorry, I saw you comment a bit too late :/ Yeah, that would be great :)

---

Yes, as I wrote above, I added that in order to do it in the common way for unix. At least I consider that as the common way. I can add an option to suppress that.

---

Let me ask: beside that you expected to have only one file, do you consider any downsides?

---

My question is:
You expect one file, ok.
You get three files, including the version pattern common in unix.
What is the downside in your opinion in having three instead of one file?

---

You are right, there is actually no real reason to change it to one file, copying over all three works fine :) Sorry for the delay, and thank you for your patience!

---

You're welcome.

---

Excellent, I guess this is resolved then.